codegen/generated!.gotpl

```diff
@@ -196,6 +208,126 @@ type executionContext struct {
 	*executableSchema
 }
 
+{{ if and .QueryDirectives .QueryRoot }}
```

Comment:
Probably need to dry this out. I suggest creating a new template called `directives.gotpl` and feed it data from a new function in directives.go. Try to paramatize out the parts that would be hard to do in the template.

Reply:
moved the template to a separate file